presto-main/src/main/java/com/facebook/presto/cost/PlanNodeCostEstimate.java

```diff
@@ -42,38 +42,91 @@ public static PlanNodeCostEstimate networkCost(double networkCost)
         return new PlanNodeCostEstimate(0, 0, networkCost);
     }
 
+    private final double individualCpuCost;
+    private final double individualMemoryCost;
+    private final double individualNetworkCost;
     private final double cpuCost;
     private final double memoryCost;
     private final double networkCost;
 
-    public PlanNodeCostEstimate(double cpuCost, double memoryCost, double networkCost)
+    /**
+     * Builds a PlanNodeCostEstimate using the individual costs for the cumulative costs
+     * PlanNodeCostEstimate becomes cumulative when the add() method is used
+     * @param individualCpuCost: the initial, local CPU cost of the PlanNode
+     * @param individualMemoryCost: the initial, local memory cost of the PlanNode
+     * @param individualNetworkCost: the initial, local network cost of the PlanNode
+     */
+    public PlanNodeCostEstimate(double individualCpuCost, double individualMemoryCost, double individualNetworkCost)
+    {
+        this(individualCpuCost, individualMemoryCost, individualNetworkCost, individualCpuCost, individualMemoryCost, individualNetworkCost);
+    }
+
+    private PlanNodeCostEstimate(double individualCpuCost, double individualMemoryCost, double individualNetworkCost,
+            double cpuCost, double memoryCost, double networkCost)
     {
-        checkArgument(isNaN(cpuCost) || cpuCost >= 0, "cpuCost cannot be negative");
-        checkArgument(isNaN(memoryCost) || memoryCost >= 0, "memoryCost cannot be negative");
-        checkArgument(isNaN(networkCost) || networkCost >= 0, "networkCost cannot be negative");
+        checkArgument(isNaN(individualCpuCost) || individualCpuCost >= 0,
+                "individualCpuCost cannot be negative");
+        checkArgument(isNaN(individualMemoryCost) || individualMemoryCost >= 0,
+                "individualMemoryCost cannot be negative");
+        checkArgument(isNaN(individualNetworkCost) || individualNetworkCost >= 0,
+                "individualNetworkCost cannot be negative");
+        checkArgument(isNaN(cpuCost) || cpuCost >= 0,
+                "cpuCost cannot be negative");
+        checkArgument(isNaN(memoryCost) || memoryCost >= 0,
+                "memoryCost cannot be negative");
+        checkArgument(isNaN(networkCost) || networkCost >= 0,
+                "networkCost cannot be negative");
+
+        this.individualCpuCost = individualCpuCost;
+        this.individualMemoryCost = individualMemoryCost;
+        this.individualNetworkCost = individualNetworkCost;
         this.cpuCost = cpuCost;
         this.memoryCost = memoryCost;
         this.networkCost = networkCost;
     }
 
     /**
-     * Returns CPU component of the cost. Unknown value is represented by {@link Double#NaN}
+     * Returns individual CPU component of the cost. Unknown value is represented by {@link Double#NaN}
+     */
+    public double getIndividualCpuCost()
```

Comment:
What does "individual" mean in this context? Individual with respect to what?

Reply:
`individual` here refers to the cost of the node itself; before this change, the estimate was tracking the cumulative cost of the whole sub-tree rooted at that node.

Reply:
In that case, I'd suggest we rename the existing methods to `getCumulativeCpuCost`, etc. and make these look like "getCpuCost". The term "individual" is confusing, whereas "cumulative" has much more familiar connotation, especially in the context of a tree.